lisa/microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -1338,6 +1426,707 @@ def _set_block_size_env_var(self, testcase: str) -> None:
+                    f"Only {allocated}GB (1GB pages) allocated (requested 16GB)"
+                )
+        else:
+            # Fallback to 2MB hugepages (8192 pages = 16GB)
+            self.node.execute(
+                f"echo 8192 | sudo tee {hugepage_2m_path} || true",
+                shell=True,
+                sudo=True,
+            )
+            # Verify allocation
+            verify = self.node.execute(
+                f"cat {hugepage_2m_path}",
+                shell=True,
+            )
+            allocated = int(verify.stdout.strip()) if verify.exit_code == 0 else 0
+            # Convert 2MiB pages to GiB
+            allocated_gib = allocated * 2 / 1024
+            if allocated >= 8192:
+                self._log.debug(f"Reserved 16GB (2MB pages) on node{self._numa_node}")
+            else:
+                self._log.debug(
+                    f"Only {allocated_gib:.2f} GiB (2MiB pages) allocated "
+                    f"(requested 16 GiB)"
+                )
+
+        # Export NUMA node for CH launcher
+        os.environ["CH_NUMA_NODE"] = str(self._numa_node)
+
+    def _run_anchor_gate(self) -> None:
+        """
+        Anchor gate: 5-8s CPU/mem warmup to validate system stability.
+
+        Validates against EWMA baseline (±5%). Retries once on failure.
+        Uses exponential weighted moving average (alpha=0.3) for stability.
+        """
+        # 5-8s CPU/mem anchor using dd
+        start = time.time()
+        self.node.execute(
+            "dd if=/dev/zero of=/dev/null bs=1M count=2048 status=none",
+            shell=True,
+            timeout=15,
+        )
+        elapsed = time.time() - start
+
+        # Calculate throughput (GB/s) - 2048 MB = 2.048 GB
+        throughput = 2.048 / elapsed
+
+        # EWMA validation (skip first run to establish baseline)
+        if self._anchor_ewma_count > 0:
+            deviation = abs(throughput - self._anchor_ewma) / self._anchor_ewma
+
+            if deviation > self.ANCHOR_DEVIATION_THRESHOLD:
+                self._log.debug(
+                    f"Anchor gate FAILED: {deviation * 100:.1f}% deviation "
+                    f"(expected: {self._anchor_ewma:.2f} GB/s, "
+                    f"got: {throughput:.2f} GB/s)"
+                )
+
+                # Retry once
+                self._log.debug("Retrying anchor gate...")
+                time.sleep(5)
+
+                start = time.time()
+                self.node.execute(
+                    "dd if=/dev/zero of=/dev/null bs=1M count=2048 status=none",
+                    shell=True,
+                    timeout=15,
+                )
+                elapsed = time.time() - start
+                throughput = 2.048 / elapsed
+
+                deviation = abs(throughput - self._anchor_ewma) / self._anchor_ewma
+
+                if deviation > self.ANCHOR_DEVIATION_THRESHOLD:
+                    self._log.debug(
+                        f"Anchor gate FAILED on retry: "
```

Comment:
What would we do with this information? This doesn't give us any actionable insights. Again, it was calculated with only 2 data points

Reply:
Today: it annotates the run. If we later see elevated CV or an odd first iteration, we can correlate that to an anchor warning during triage.